mne/beamformer/tests/test_resolution_matrix.py

```diff
@@ -0,0 +1,88 @@
+import numpy as np
+from numpy.testing import assert_
+
+import mne
+from mne.datasets import testing
+from mne.beamformer import make_lcmv, make_lcmv_resolution_matrix
+
+data_path = testing.data_path(download=False)
+subjects_dir = op.join(data_path, 'subjects')
+fname_inv = op.join(data_path, 'MEG', 'sample',
+                    'sample_audvis_trunc-meg-eeg-oct-6-meg-inv.fif')
+fname_evoked = op.join(data_path, 'MEG', 'sample',
+                       'sample_audvis_trunc-ave.fif')
+fname_raw = op.join(data_path, 'MEG', 'sample', 'sample_audvis_trunc_raw.fif')
+fname_fwd = op.join(data_path, 'MEG', 'sample',
+                    'sample_audvis_trunc-meg-eeg-oct-4-fwd.fif')
+fname_cov = op.join(data_path, 'MEG', 'sample', 'sample_audvis_trunc-cov.fif')
+
+
+@testing.requires_testing_data
+def test_resolution_matrix_lcmv():
+    """Test computation of resolution matrix for LCMV beamformers."""
+    # read forward solution
+    forward = mne.read_forward_solution(fname_fwd)
+
+    # remove bad channels
+    forward = mne.pick_channels_forward(forward, exclude='bads')
+
+    # forward operator with fixed source orientations
+    forward_fxd = mne.convert_forward_solution(forward, surf_ori=True,
+                                               force_fixed=True)
+
+    # noise covariance matrix
+    noise_cov = mne.read_cov(fname_cov)
+
+    # evoked data for info
+    evoked = mne.read_evokeds(fname_evoked, 0)
+    evoked = evoked.pick_types(meg=True, eeg=True, exclude='bads')
+
+    # Resolution matrix for Beamformer
+
+    info = evoked.info
+    # over-regularise for testing (see below)
+    noise_cov = mne.cov.regularize(noise_cov, info, mag=0.05, grad=0.05,
+                                   eeg=0.05, rank=None)
+
+    data_cov = noise_cov.copy()  # to test a property of LCMV
+
+    # compute beamformer filters
+    # reg=0. to make sure noise_cov and data_cov are as similar as possible
+    filters = make_lcmv(info, forward_fxd, data_cov, reg=0.,
+                        noise_cov=noise_cov,
+                        pick_ori=None, rank=None,
+                        weight_norm=None,
+                        reduce_rank=False,
+                        verbose=False)
+
+    resmat_lcmv = make_lcmv_resolution_matrix(filters, forward_fxd, info)
+
+    # for noise_cov==data_cov, the filter weights should be the transpose of
+    # leadfield
+
+    # create filters with transposed whitened leadfield as weights
+    filters_lfd = filters.copy()
+    filters_lfd['weights'] = forward_fxd['sol']['data'].T
+
+    # compute resolution matrix for filters with transposed leadfield
+    resmat_fwd = make_lcmv_resolution_matrix(filters_lfd, forward_fxd, info)
+
+    # Tests
+
+    # This correlation should be exactly 1.0, but it isn't
+    # probably due to different treatment of noise_cov and data_cov under the
+    # hood.
+    for v in [10, 100, 100]:  # arbitrary vertices
+        assert_(np.corrcoef(resmat_fwd[v:], resmat_lcmv[v, :])[1, 1] > .95)
```

Comment:
I think that this test does not check what you want it to check. The diagonal entries of `corrcoef` will always be `1.` based on how it represents the results. I'm going to push a commit to make it test the correlation coefficient for all rows. Unfortunately it's not anywhere near 1., even allowing numbers as low as 0.5 fail. Hopefully you'll be able to figure out why...?

Reply:
I got it to "work" with a tol of 0.12 (corr > 0.88) by using a diag cov, see latest push

Reply:
I suspect that there is a discrepancy between the way the noise covariance and data covariance matrices are inverted for the whitening and beamformer, respectively. I'll have a closer look.

Reply:
@larsoner I've changed the test for LCMV beamformer resolution matrix again. I'm now checking that all rows of the resolution matrices for whitened LCMV and transposed leadfield are almost perfectly correlated - and they are, at least for the diagonal noise covariance matrix. I think previous discrepancies for real noise covariance matrices were due to regularisation. I'll look into this, but for now I think this test does its job.